src/cli/generator/config.rs

```diff
@@ -139,11 +139,18 @@ impl Output<UnResolved> {
 }
 
 impl Source<UnResolved> {
-    pub fn resolve(self, parent_dir: Option<&Path>) -> anyhow::Result<Source<Resolved>> {
+    pub fn resolve(
+        self,
+        parent_dir: Option<&Path>,
+        reader_context: &ConfigReaderContext,
+    ) -> anyhow::Result<Source<Resolved>> {
         match self {
             Source::Curl { src, field_name, headers } => {
+                // Resolve the header values with mustache template.
+                let resolved_headers =
+                    headers.and_then(|headers_inner| headers_inner.resolve(reader_context).ok());
```

Comment:
`.ok` here hides possible error. You probably want to propagate the error in that case. 

I just leave it here https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose